code/game/objects/items/weapons/signs.dm

```diff
@@ -0,0 +1,34 @@
+/obj/item/weapon/picket_sign
+	icon_state = "picket"
+	name = "blank picket sign"
+	desc = "It reads:"
+
+	force = 5
+	w_class = 4.0
+
+	attack_verb = list("bashed","smacked")
+
+	var/label = ""
+
+/obj/item/weapon/picket_sign/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/pen) || istype(W, /obj/item/toy/crayon))
+		var/txt = stripped_input(user, "What would you like to write on the sign?", "Sign Label", null , 30)
+		if(txt)
+			label = txt
+			src.name = "[label] sign"
+			desc =	"It reads: [label]"
+	..()
+
+/obj/item/weapon/picket_sign/attack_self(mob/living/carbon/human/user)
+	if(label!="")
```

Comment:
Leading whitespace? What? Your check here simply reads "If label is not empty" which is the same as ```if(label)```, which is just cleaner.